167 appendix c use cases (#184)
IHE/sdpi-fhir: SDPi_Supplement/asciidoc/volume1/tf1-ch-a-requirements-interoperability.adoc

[sdpi_offset=1]
=== Requirements:  From Narratives to Plug-and-Trust Interfaces

For every device or system interface that supports the elements specified in this technical framework, requirements and capabilities are derived from many different standards, types of standards, and standards development organizations (SDO).  Some are technology focused, such as the ISO/IEEE 11073 standards for health device communication, and others are more focused on quality and risk management, such as the ISO/IEC 80001-1 and ISO 14971 standards.  They are all profiled and implemented and tested in a single interface point, and thus need to be integrated in a coherent and cohesive manner.

> In order to manage this diversity and wealth of specifications and standards, the following <<hanging_gardens_framework>> graphic was created:

Reviewer: Generally, I prefer refering to tables, listings and graphics by using Asciidoc references over using "the following".

If we don't use numbering for reference:
`In order to manage this diversity and wealth of specifications and standards, the <<hanging_gardens_framework>> graphic was created.`

Or if we do use numbering for references: 

`In order to manage this diversity and wealth of specifications and standards, the Hanging Gardens Framwork as specified in Figure <<hanging_gardens_framework>> was created.`

Reviewer: The `<<hanging_gardens_framework>>` reference does not exist.

.Hanging Gardens Framework

> 

Reviewer: ```suggestion
[#hanging_gardens_framework]
```

Author: Agree on the need to generally avoid "The following ...", though in this case - it is pretty obvious.  That said, how would you integrate in-line text for the autoreferences for Figure 1. Hanging Gardens Framework?  Which is defined (vs. defining a [#...] anchor.  In other words, how can I call out the text of the reference "Hanging Gardens Framework" and separately the number "Figure 1"?    

Similar to :xrefstyle" basic vs. short ... only with inline text vs. document attributes.

Reviewer: No clue and it will be subject to change anyway - because of our custom numbering...

It should be sufficient to just refer to <<...>> without taking care of any details. Replacing the reference by actual text should be the job of the asciidoc compiler/converter so that each reference is transformed uniformly.

image::../images/vol1-diagram-hanging-gardens.svg[]

Though a full explanation of the this framework is beyond this discussion footnote:hanging_gardens_framework_article[A more detailed explanation of this model is provided on the https://confluence.hl7.org/x/4ijxB[IHE-HL7 Gemini _Hanging Gardens Framework_ confluence page].  Last accessed 2022.10.04.], some general observations will help understand the value and use of the framework:

> . Real-world Requirements are at the top -- use case requirements, detailing clinical system functions ensure that all requirements and capabilities are rooted in the purposes that technology users are advancing, namely to provide safe

Reviewer: This bullet point seems to introduce a sub-list but other bullet points on the same indentation follow. Or the end of the sentence is missing.

> . <inter layer interfaces Requirements -> Capabilities API, Req Def -> Req Usage >

Reviewer: This looks like preliminary information - maybe it makes sense to provide a todo or an issue that covers provision of a final text here.

Author: Could do that and then finalize in #142, though I'd have to rebase that given how long it has been out there!  The integration of use cases into specific profiles + the updating of the use case specifications with the appropriate metadata could happen in other branches.

. <Standards groups / families - internally coherent ... hopefully!>
. <Vertical / horizontal>
. <profiles navigate from the top to the bottom similar with each layer contributing to those below and converging in a single interface capability bundle>
. <physical layers are leveraged / not re-invented!>
. <requirements interoperability traces "profile lines" from top to bottom>
. <requirements TYPEs are rooted in the layers that they represent and link Defs to Usage, see ##TODO:  ADD LINK HERE to "SDPi Requirements Core Model"" ## section below.>

#TODO:  ADD INTEROPERABILITY PATHS EXAMPLE VIA A TABLE (VS. LINES THROUGH GRAPHIC)#

> Note:  The implementation of this framework will be extended as the specifications and tooling mature.

Reviewer: Is there a reason why this note was not provided with the `NOTE:` directive?

```suggestion
NOTE: The implementation of this framework will be extended as the specifications and tooling mature.
```

During the development of the 80001-1 standard, though, it was recognized that risk management is just one of a number of other core themes that had to be managed in concert (e.g., quality management, human factors / usability).  Also ensuring SES required processes that involve a total product lifecycle, with responsibilities transitioning across that period.  To address these requirements, JWG7 developed the <<iso_81001_1_2021>> standard, which also included The Temple diagram to communicate the various aspects of SES that must be considered and managed:

.ISO/IEC 81001-1 _The Temple Diagram_

> 

Reviewer: ```suggestion
[#the_temple_diagram]
```

Author: See similar Reply question to Hanging Gardens Comment above.  Whatever is done there can be replicated here.

image::../images/vol1-diagram-81001-temple.svg[algin=center]

> #TODO:  SHOULD THE FOLLOWING SOURCE LINE BE A FOOTNOTE ... HUNG ON WHERE ... IHE CONVENTION?#

Author: What is unique about this image is that there is no "source" (e.g., puml, pptx) but the SVG is downloaded directly from the site referenced in Appendix B <<iso_81001_temple>>.  To that end, I could just leave off the image Source, or is there a better way of incorporating this into the metadata itself?  For example, 

.A mountain sunset
[#img-sunset,link=https://www.flickr.com/photos/javh/5448336655]
image::sunset.jpg[Sunset,200,100]

per documentation.

[none]
. Source:  <<iso_81001_temple>>

> One of the key challenges for implementing this standard, though, is what might be labeld:  *_The Interoperability Trust Gap_*.  This is the technology "hand off" space between the left side of the lifecycle -- Design and development phase, where key responsibility is by each of the "Accountable Manufacturer" organizations, and the right side of the lifecycle -- Implementation phase & Clinical use phase, where key responsibility is on the Accountable Healthcare Delivery Organization (HDO).  Though this reads well in the standards and the model organizes everything in a clear fashion, operationalizing this in real world use remains a Sisyphean effort, primarily due to the amount of expertice, time and resources needed to effectively implement the SES standards as part of normal operating business in HDOs.

Reviewer: ```suggestion
One of the key challenges for implementing this standard, though, is what might be labeled:  *_The Interoperability Trust Gap_*.  This is the technology "hand off" space between the left side of the lifecycle -- Design and development phase, where key responsibility is by each of the "Accountable Manufacturer" organizations, and the right side of the lifecycle -- Implementation phase & Clinical use phase, where key responsibility is on the Accountable Healthcare Delivery Organization (HDO).  Though this reads well in the standards and the model organizes everything in a clear fashion, operationalizing this in real world use remains a Sisyphean effort, primarily due to the amount of expertise, time and resources needed to effectively implement the SES standards as part of normal operating business in HDOs.
```

To address this SES implementation problem, the SDPi profiles:

> . Leverage the ISO/IEEE 11073-1070x Participant Key Purposes (PKP) standards, which represent a consensus standard for risk management of technologies that are implemented in that left-right gap on the Temple model

Reviewer: PKP, ICS, RI, SES, MDI and other abbrevs should be introduced in the glossary and linked in the text via `<<acronym_[...]>>`

> . Implementation Conformance Statement (ICS) tables from each of these PKP standards is included in TF-1 Appendix B of this specification, with indication as to whether, how and where each requirement is addressed

Reviewer: `TF-1 Appendix B` should be linked accordingly.
This bullet point is missing a full stop opposed to the next two bullet points.

. "Safety, Effectiveness, Security Considerations and Requirements" sections are integrated throughout the profile specifications to link from the PKP ICS table requirements to the satisfying capabilities.
[none]

> .. Additional non-PKP risk management will also be performed by subject matter experts and formalized in these SES consideration sections, where appropriate.

Reviewer: ```suggestion
.. Additional non-PKP risk management will also be performed by subject matter experts and formalized in these SES Consideration sections, where appropriate.
```

> These SES Considerations sections are the result of the IHE "Security Considerations" sections + the IHE Devices "Safety Considerations" sections, but now consolidated into a single SES Considerations section that includes the 3rd risk management property, Effectiveness, and is tied to meeting the requirements of specific standards.

Reviewer: `SES Considerations` or `SES considerations` - should be consistent accross the text

> NOTE:  The moniker *_SES+MDI_* is shorthand to refer to the integration of the technical medical device interoperability (MDI) specifications  with the application of quality / risk management SES standards and processes.

Reviewer: Also a good place to move that to the glossary?

> How does this address the "interoperability trust gap"?  By integrating SES directly into the specifications, especially integrating the ISO/IEC 11073-1070x standards, enabling "plug-and-trust" system product components, the SES implementation and operational requirements and responsibilities are greatly reduced, the "gap" is filled for all stakeholders, and the the goals of improved safety, security and clinical effectiveness of technology can be readily realized.

Reviewer: ```suggestion
How does this address the "interoperability trust gap"?  By integrating SES directly into the specifications, especially integrating the ISO/IEC 11073-1070x standards, enabling "plug-and-trust" system product components, the SES implementation and operational requirements and responsibilities are greatly reduced, the "gap" is filled for all stakeholders, and the goals of improved safety, security and clinical effectiveness of technology can be readily realized.
```

=== SDPi Requirements Modeling & Integration

> As pointed out above, requirements interoperability (RI) based on robust model-based metadata is a core, innovative aspect of this SDPi profiles specification.  Given the ultimate intent to realize this description as a _Model Centric (MC) single-source-of-truth, computable, simulatable, verifiable and validatable system of systems interoperability specification_, and recognizing that it will take a significant transition period from a document-centric approach to a model-centric approach, a simplified requirements model is provided below but is aligned with the <<omg_sysml_2-0_spec>> section 7.20 Requirements language.  Of course, that specification provides for significantly more detailed and complext modeling, the general constructs may be used in this document to start the transition toward that model.  Note that SysML 2.0 also better supports model interoperability (tool-independent model exchange) and _Model-based Systems Engineering (MBSE)_ (see https://en.wikipedia.org/wiki/Model-based_systems_engineering[MBSE Wikipedia article and references]), as well as the <<ihe_eu_experience_2021_presentation_cooper_schlichting>> for an overview presentation of MBSE, MedTech system V&V, and IHE Conformity Assessment.

Reviewer: ```suggestion
As pointed out above, requirements interoperability (RI) based on robust model-based metadata is a core, innovative aspect of this SDPi profiles specification.  Given the ultimate intent to realize this description as a _Model Centric (MC) single-source-of-truth, computable, simulatable, verifiable and validatable system of systems interoperability specification_, and recognizing that it will take a significant transition period from a document-centric approach to a model-centric approach, a simplified requirements model is provided below but is aligned with the <<omg_sysml_2_0_spec>> section 7.20 Requirements language.  Of course, that specification provides for significantly more detailed and complex modeling, the general constructs may be used in this document to start the transition toward that model.  Note that SysML 2.0 also better supports model interoperability (tool-independent model exchange) and _Model-based Systems Engineering (MBSE)_ (see https://en.wikipedia.org/wiki/Model-based_systems_engineering[MBSE Wikipedia article and references]), as well as the <<ihe_eu_experience_2021_presentation_cooper_schlichting>> for an overview presentation of MBSE, MedTech system V&V, and IHE Conformity Assessment.
```

Reviewer: A general rule of thumb for those text-based and Git-based document development: 1 sentence per line increases readability in the text editor und display of diffs. 

```
As pointed out above, requirements interoperability (RI) based on robust model-based metadata is a core, innovative aspect of this SDPi profiles specification.
Given the ultimate intent to realize this description as a _Model Centric (MC) single-source-of-truth, computable, simulatable, verifiable and validatable system of systems interoperability specification_, and recognizing that it will take a significant transition period from a document-centric approach to a model-centric approach, a simplified requirements model is provided below but is aligned with the <<omg_sysml_2_0_spec>> section 7.20 Requirements language.
Of course, that specification provides for significantly more detailed and complex modeling, the general constructs may be used in this document to start the transition toward that model.
Note that SysML 2.0 also better supports model interoperability (tool-independent model exchange) and _Model-based Systems Engineering (MBSE)_ (see https://en.wikipedia.org/wiki/Model-based_systems_engineering[MBSE Wikipedia article and references]), as well as the <<ihe_eu_experience_2021_presentation_cooper_schlichting>> for an overview presentation of MBSE, MedTech system V&V, and IHE Conformity Assessment.
```

Single line breaks will not create a line break in the output format.

Author: Updated

> It should be further noted that though conformity testing aspects are beyond this SDPi specification, the modeling constructs used below will also be integrated with <<omg_sysml_2-0_spec>> section 7.23 Verification Cases, to provide for advanced V&V of interoperable system components and entire systems of products.

Reviewer: ```suggestion
It should be further noted that though conformity testing aspects are beyond this revision of the SDPi specification, the modeling constructs used below will also be integrated with <<omg_sysml_2-0_spec>> section 7.23 Verification Cases, to provide for advanced V&V of interoperable system components and entire systems of products.
```

Author: Updated.

==== SDPi Requirements Core Model

To formally integrate requirements in to this specification, the following requirements model provides the starting point:

.SDPi Requirements - Core Model

image::../images/vol1-diagram-sdpi-req-types-model.svg[align=center]

This model identifies a set of requirement "types" that are formalized in the specification.  Each type is a source of requirements that are explicitly identified and formalized with appropriate metadata.

[%autowidth]
[cols="^1,4,^1,^1"]
|===
|Model Element |Description |AsciiDoc Attribute |Further Specified

| SDPi Requirement

> | A defined stakeholde-imposed constraints that must be satisfied for a design solution to be valid.  This is an {abstract} element.

Reviewer: What do you mean by `{abstract}`?

> | A defined stakeholde-imposed constraints that must be satisfied for a design solution to be valid.  This is an {abstract} element.

Author: Updated.  For the UML model, the top object is never instantiated.  So ... standard UML use of the term {abstract}.  In IHE, there are varying conventions for this - like the SOMDS_PARTICIPANT abstract actor in the SDPi-P actor model.  It is only realized as either a provider or a consumer.  For the work that you're doing in TF-2, this would be addressed by the "Tech Feature" metadata spec, which I'll probably include in this section. OK?

> | A defined stakeholde-imposed constraints that must be satisfied for a design solution to be valid.  This is an {abstract} element.

Author: Updated the content for clarity:  **_This is an {abstract} class model element._**

| sdpi_requirement
| See subtypes

| SDPi Requirement Group
| Two or more SDPi Requirements may be collected into a group that is focused around a specific _subject_ area.
| sdpi_requirement_group
|

| Usage
| Requirement utilized in a specific use context that provides for its satisfaction.
| sdpi_requirement_usage
| 

| Use Case Feature
| A functional "feature" requirement based on clinical use case scenarios.
| mdi_requirement_use_case
| See TF-1 Appendix C, gherkin model

| Ref. Standard ICS
| Requirement definitions that are specified in a normative reference.
| sdpi_requirement_ref_standard
|

| SES
|
| sdpi_requirement_ses

> | See SES section ... WHERE IS THAT?

Reviewer: Add a todo that this table needs some work?

Author: This is true of the entire TF-1 Appendix A ... which is a work-in-progress, to be finished before integration into master.
